server/meshmodel/helper.go

```diff
@@ -0,0 +1,135 @@
+package meshmodel
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/layer5io/meshery/server/helpers/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/meshmodel"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+const ArtifactHubComponentsHandler = "kubernetes" //The components generated in output directory will be handled by kubernetes
+
+type ComponentHelper struct {
+	handlerConfig    *models.HandlerConfig
+	regManager       *meshmodel.RegistryManager
+	componentChan    chan v1alpha1.ComponentDefinition
+	relationshipChan chan v1alpha1.RelationshipDefinition
+	doneSignal       chan bool
+	errorChan        chan error
+}
+
+func (ch *ComponentHelper) Init(hc *models.HandlerConfig, rm *meshmodel.RegistryManager) {
+	ch.handlerConfig = hc
+	ch.regManager = rm
+	ch.componentChan = make(chan v1alpha1.ComponentDefinition, 1)
+	ch.relationshipChan = make(chan v1alpha1.RelationshipDefinition, 1)
+	ch.doneSignal = make(chan bool)
+	ch.errorChan = make(chan error)
+}
+
+// seed the local meshmodel components
+func (ch ComponentHelper) SeedComponents() {
+	// Read component and relationship definitions from files and send them to channels
+	ch.generateComponents("/components")
+	ch.generateRelationships("/relationships")
+
+	// Register components and relationships with the registry manager
+	go ch.watchComponents()
```

Comment:
@Azanul Shouldn't this be called before generateComponentst? generateComponents is a synchronous call that will block on the componentChannel after pushing first component until the component is received. And unless you run watchcomponent, it wont be received.

Reply:
You're right, my bad. I made them async first but then removed go and didn't notice this. How about executing both `generateComponents` and `generateRelationships` in their own go routines.